### PR TITLE
Update xtb 6.6.0

### PR DESCRIPTION
**Pull request by awvwgk**
Released at https://github.com/grimme-lab/xtb/releases/tag/v6.6.0

### CODE REVIEW

File: var/spack/repos/builtin/packages/tblite/package.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/tblite/package.py]:
```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

File: var/spack/repos/builtin/packages/tblite/package.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
+class Tblite(MesonPackage):
+    """Light-weight tight-binding framework"""
+
+    homepage = "https://tblite.readthedocs.io"
+    url = "https://github.com/tblite/tblite/releases/download/v0.3.0/tblite-0.3.0.tar.xz"
+    git = "https://github.com/tblite/tblite.git"
+
+    maintainers = ["awvwgk"]
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/tblite/package.py]: The syntax for adding maintainers has changed.  See https://spack.readthedocs.io/en/latest/packaging_guide.html#maintainers

```suggestion
    maintainers("awvwgk")
```

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256 for both packages.  Just a couple of nits.

**Review by awvwgk — COMMENTED** [review]
Apply suggested changes

**Review by haampie — APPROVED** [review]

File: var/spack/repos/builtin/packages/xtb/package.py [hunk 3]
```diff
@@ -32,7 +34,9 @@ def meson_args(self):
         elif lapack != "openblas":
             lapack = "netlib"
 
+        lapack_opt = "-Dlapack={0}" if self.version > Version("6.6.0") else "-Dla_backend={0}"
```
Comment (ltalirz) [inline var/spack/repos/builtin/packages/xtb/package.py:37]: Building 6.6.0 with this recipe fails with
```
>> 9    ../spack-src/meson.build:19:0: ERROR: Unknown options: "la_backend"
```


Looking at https://github.com/grimme-lab/xtb/pull/241/files it appears to me that `la_backend` is to be _introduced_ in 6.6.1.

That would mean this logic is incorrect, and it should rather be
```
lapack_opt = "-Dlapack={0}" if self.version <= Version("6.6.0") else "-Dla_backend={0}"
```

Is that correct?

Comment (awvwgk) [reply var/spack/repos/builtin/packages/xtb/package.py:37]: Could you open a new issue or send a PR fixing this? Commenting in a merged PR is likely to get overlooked.

Comment (ltalirz) [reply var/spack/repos/builtin/packages/xtb/package.py:37]: https://github.com/spack/spack/pull/38309

**Pull request merged**